File: dev_board_test/Batch_Plot.cc
 //define string variables
TString runname = "l6b_axial_hole_f7_h_b180v_nominal";
TString dataname = "sample";
TString str1;
TString str2;

//load file
f = new TFile("20170830_"+runname+"_tuple.root");

//create canvas
TCanvas *c1 = new TCanvas("c1","sample vs pchannel",800,600);
c1->Print(runname+".pdf[")

//plot 2D histo
for (Int_t i=0;i<6;i++) {
str1.Form(dataname+"%d:pchannel>>hist1(700,0,700,2500,4500,7000)",i);
str2.Form(runname+"_sample%d",i);
results->Draw(str1,"","COLZ");
c1->Update();
Printf("i=%d",i);
c1->Update();
c1->Print(runname+".pdf");
}

c1->Print(runname+".pdf]");

File: Code/dev_board_test/Batch_Plot.py
import os
path = '/home/nanw/HPS/dev_board_test'

import ROOT
from ROOT import TFile, TCanvas, TTree, gROOT

#create arrays
fn = []
histarray = []

#define canvas
c1 = TCanvas("c1","sample vs channel",800,600);

for filename in os.listdir(path):
	if filename.endswith(".root"):
		fn.append(filename)
	else:
		continue

for i in range(len(fn)):
	runname = fn[i].replace(".root","")
	#c1.Print(runname+".pdf[")
	f = TFile(path+'/'+fn[i])
	events = f.Get("results")
	for ii in range(6):
		histname = 'hist'+str(ii)
		str1 = 'sample'+str(ii)+':pchannel>>'+histname+'(700,0,700,2500,4500,7000)'
		events.Draw(str1)
		histo = gROOT.FindObject(histname)
		histarray.append(histo)
		print histname
		print histarray[0]
		print histarray[i*6+ii]
		#c1.Print(runname+".pdf")
	#c1.Print(runname+".pdf]")
	del f, events

